nulib/devstack
# Update main branch to v2.0.0

I've been using a different branch of `devstack` for a little while now, to make sure it would be rock solid when the time came for everyone to use it. This PR merges it into `main` and makes it available via `devstack update`.

## Major Changes

* Uses the `docker compose` command which is now built into Docker instead of the separate `docker-compose` script
* Adds `devstack utils` to make it easy (but optional) to install some utility scripts to work with AWS resources
* Adds a fixity/checksum webhook to minio so that certain buckets can mimic the auto-tagging behavior of the [Serverless Fixity for Digital Preservation Compliance](https://github.com/awslabs/serverless-fixity-for-digital-preservation-compliance) step function
* Removes obsolete configurations for container images we no longer use (e.g., `sqs` and `pgadmin`) as well as application sets we no longer need (`donut`) 

## How to Test

1. `devstack down -v`
2. `devstack update`
3. cd to the devstack working directory
4. `git switch use-internal-docker-compose`
5. `devstack version` (should give you `devstack version 2.0.0`)
6. Go do some normal devstack things – bring up the stack(s), run app tests, etc.

Some things will look different (because of the new `docker compose` command), but everything should work as expected.
